UnexpectedException: nameModule

When editing a file, it went away after a few more edits. The bit of code I was editing was:

```
instance Arbitrary File where
    arbitrary = arbitraryBoundedEnum 
```

Can't seem to reproduce it.

```
[DEBUG] Finishing shakeRun (took 0.08s, exception: Error when running Shake build system:
ghcide.exe: panic! (the 'impossible' happened)
  (GHC version 8.6.5 for x86_64-unknown-mingw32):
	nameModule

system arbitrarySizedBoundedEnum_aB1J
Call stack:
    CallStack (from HasCallStack):
      callStackDoc, called at compiler\utils\Outputable.hs:1160:37 in ghc:Outputable
      pprPanic, called at compiler\basicTypes\Name.hs:240:3 in ghc:Name

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug

)
[ERROR] Unexpected exception on request, please report!
Message: RequestMessage {_jsonrpc = "2.0", _id = IdInt 1546, _method = TextDocumentDefinition, _params = TextDocumentPositionParams {_textDocument = TextDocumentIdentifier {_uri = Uri {getUri = "file:///c%3A/Neil/rattle-papers/model/Main.hs"}}, _position = Position {_line = 35, _character = 30}, _workDoneToken = Nothing}}
Exception: Error when running Shake build system:
ghcide.exe: panic! (the 'impossible' happened)
  (GHC version 8.6.5 for x86_64-unknown-mingw32):
	nameModule

system arbitrarySizedBoundedEnum_aB1J
Call stack:
    CallStack (from HasCallStack):
      callStackDoc, called at compiler\utils\Outputable.hs:1160:37 in ghc:Outputable
      pprPanic, called at compiler\basicTypes\Name.hs:240:3 in ghc:Name

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug


[Error - 11:50:16] Request textDocument/definition failed.
  Message: Error when running Shake build system:
ghcide.exe: panic! (the 'impossible' happened)
  (GHC version 8.6.5 for x86_64-unknown-mingw32):
	nameModule

system arbitrarySizedBoundedEnum_aB1J
Call stack:
    CallStack (from HasCallStack):
      callStackDoc, called at compiler\utils\Outputable.hs:1160:37 in ghc:Outputable
      pprPanic, called at compiler\basicTypes\Name.hs:240:3 in ghc:Name

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug


  Code: -32603 
```

--- a/.hlint.yaml
+++ b/.hlint.yaml
@@ -97,9 +97,12 @@
 #   - {name: Control.Arrow, within: []} # Certain modules are banned entirely
 #
 - functions:
+  # Things that are unsafe in Haskell base library
   - {name: unsafeInterleaveIO, within: []}
   - {name: unsafeDupablePerformIO, within: []}
   - {name: unsafeCoerce, within: []}
+  # Things that are a bit dangerous in the GHC API
+  - {name: nameModule, within: []}
 
 # Add custom hints for this project
 #

--- a/src/Development/IDE/Spans/AtPoint.hs
+++ b/src/Development/IDE/Spans/AtPoint.hs
@@ -133,7 +133,7 @@ locationsAtPoint getHieFile IdeOptions{..} pkgState pos =
                 -- This case usually arises when the definition is in an external package.
                 -- In this case the interface files contain garbage source spans
                 -- so we instead read the .hie files to get useful source spans.
-                let mod = nameModule name
+                mod <- MaybeT $ return $ nameModule_maybe name
                 let unitId = moduleUnitId mod
                 pkgConfig <- MaybeT $ pure $ lookupPackageConfig unitId pkgState
                 hiePath <- MaybeT $ liftIO $ optLocateHieFile optPkgLocationOpts pkgConfig mod
@@ -147,7 +147,7 @@ locationsAtPoint getHieFile IdeOptions{..} pkgState pos =
                 pure span
         -- We ignore uniques and source spans and only compare the name and the module.
         eqName :: Name -> Name -> Bool
-        eqName n n' = nameOccName n == nameOccName n' && nameModule n == nameModule n'
+        eqName n n' = nameOccName n == nameOccName n' && nameModule_maybe n == nameModule_maybe n'
         setFileName f (RealSrcSpan span) = RealSrcSpan (span { srcSpanFile = mkFastString f })
         setFileName _ span@(UnhelpfulSpan _) = span
 

